worlds/khrecom/Regions.py

```diff
@@ -0,0 +1,379 @@
+from typing import Dict, List, NamedTuple, Optional
+
+from BaseClasses import MultiWorld, Region, Entrance
+from .Locations import KHRECOMLocation, location_table, get_locations_by_category
+
+
+class KHRECOMRegionData(NamedTuple):
+    locations: Optional[List[str]]
+    region_exits: Optional[List[str]]
+
+
+def create_regions(multiworld: MultiWorld, player: int, options):
+    regions: Dict[str, KHRECOMRegionData] = {
+        "Menu":     KHRECOMRegionData(None, ["Traverse Town"]),
+        "Traverse Town":  KHRECOMRegionData([], ["Castle Oblivion Halls"]),
+        "Wonderland":  KHRECOMRegionData([], []),
+        "Olympus Coliseum":  KHRECOMRegionData([], []),
+        "Monstro":  KHRECOMRegionData([], []),
+        "Agrabah":  KHRECOMRegionData([], []),
+        "Halloween Town":  KHRECOMRegionData([], []),
+        "Atlantica":  KHRECOMRegionData([], []),
+        "Neverland":  KHRECOMRegionData([], []),
+        "Hollow Bastion":  KHRECOMRegionData([], []),
+        "100 Acre Wood": KHRECOMRegionData([], []),
+        "Twilight Town": KHRECOMRegionData([], []),
+        "Destiny Islands": KHRECOMRegionData([], []),
+        "Castle Oblivion": KHRECOMRegionData([], []),
+        "Heartless":KHRECOMRegionData([], []),
+        "Levels":KHRECOMRegionData([], []),
+        "Castle Oblivion Halls":     KHRECOMRegionData([], ["Wonderland", "Olympus Coliseum", "Monstro",
+                                         "Agrabah", "Halloween Town", "Atlantica",
+                                         "Neverland", "Hollow Bastion", "100 Acre Wood",
+                                         "Twilight Town", "Destiny Islands", "Castle Oblivion", "Heartless", "Levels"])
+    }
+
+    # Set up locations
+
+    regions["Traverse Town"].locations.append("Starting Checks (Attack Cards Kingdom Key)")
+    regions["Traverse Town"].locations.append("Starting Checks (Item Cards Potion)")
+    regions["Traverse Town"].locations.append("Starting Checks (Magic Cards Blizzard)")
+    regions["Traverse Town"].locations.append("Starting Checks (Magic Cards Cure)")
```

Comment:
Any specific reason you're going this route instead of just putting them all up in your regions dict?
```py
"Traverse Town": KHRECOMRegionData(["Starting Checks (Attack Cards Kingdom Key)", ..., ["..."]),
```

Or potentially making a second list, so you can add one list to the other.
```py
regions["Traverse Town"].locations.extend(["Traverse Town Room of Guidance", "Traverse Town Room of Truth", ...])
```

Reply:
No particular reason, this is just the way I saw it done on Rogue Legacy which is the basis for my apworlds so I just copied their structure.

Reply:
Fair enough lol
Would recommend for the future to do it like one of those two options suggested (or refactoring this one to do that if you feel like it), but it's fine as-is